### PR TITLE
fix: do not show publish button when no ingredients

### PR DESCRIPTION
**Pull request by stephanegigandet**
Hide the "Publish" button if there are no ingredients.

Currently it does not work:

- When the app starts, the button is not shown
- Adding an ingredient makes the button show
- Deleting the ingredient does not hide the button

The Nutri-Score is updated, but not the shared button...

image

### CODE REVIEW

**Comment by alexfauquette** [comment]
Because the ingredients list is not empty, it contains an ingredient with no quantity. This should solve your issue

I added it in a commit
```diff
diff --git a/redux/reducers/recipes.ts b/redux/reducers/recipes.ts
index 97dd85b..06a5980 100644
--- a/redux/reducers/recipes.ts
+++ b/redux/reducers/recipes.ts
@@ -257,6 +257,7 @@ const ingredientReducer = (
     );
     return ingredients;
   }
+  if (action.type === "delete") {
   const { ingredientId, quantityId } = action;
 
   const ingredientIndex = ingredients.findIndex(
@@ -266,16 +267,23 @@ const ingredientReducer = (
     return ingredients;
   }
 
+    const newQuantities = ingredients[ingredientIndex].quantities.filter(
+      ({ id }) => quantityId !== id
+    );
   return [
     ...ingredients.slice(0, ingredientIndex),
+      ...(newQuantities.length === 0
+        ? []
+        : [
     {
       ...ingredients[ingredientIndex],
-      quantities: ingredients[ingredientIndex].quantities.filter(
-        ({ id }) => quantityId !== id
-      ),
+              quantities: newQuantities,
     },
+          ]),
     ...ingredients.slice(ingredientIndex + 1),
   ];
+  }
+  return ingredients;
 };
 ```

File: components/PublishRecipe.tsx [hunk 1]
```diff
@@ -98,23 +99,19 @@ const PublishRecipe = () => {
...
+      </Button>
     );
+  } else {
+    return null;
   }
-  else {
-    return;
-  }
-
 };
```
Comment (alexfauquette) [inline components/PublishRecipe.tsx]: A component should always return something. If there is nothing to render, it should be `null`

**Comment by stephanegigandet** [comment]
> Because the ingredients list is not empty, it contains an ingredient with no quantity. This should solve your issue

Thank you!

File: components/PublishRecipe.tsx (hunk 1 above)
Comment (stephanegigandet) [reply components/PublishRecipe.tsx]: OK, thanks.

**Comment by stephanegigandet** [comment]
@alexfauquette I added a few more changes to display the 0 to 100 score for Nutri-Score, and to hide the initial Nutri-Score if we don't have one. Could you review and approve the PR so that we can merge? Thank you!

File: components/PublishRecipe.tsx [hunk 2]
```diff
@@ -11,12 +11,15 @@ import { useTranslation } from "react-i18next";
 import data from "../data";
 
 const PublishRecipe = () => {
-  const [open, setOpen] = React.useState(false);
+  console.log("PublishRecipe");
```
Comment (alexfauquette) [inline components/PublishRecipe.tsx]: I assume it's for debugging
```suggestion
```

File: components/ShowNutritionalTable.tsx [hunk 3]
```diff
@@ -78,10 +78,14 @@ const EvolutionIcon = ({
 };
 
 export default function ShowNutritionalTable() {
+  console.log("ShowNutritionalTable");
```
Comment (alexfauquette) [inline components/ShowNutritionalTable.tsx]:
```suggestion
```

**Review by alexfauquette — APPROVED** [review]

**Pull request merged**